src/frontend/src/components/LogIn/index.tsx

```diff
@@ -1,61 +1,173 @@
 import React from 'react';
 import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
 import { AppState } from '../../redux/store';
 import { logIn } from '../../redux/system/actions';
-import Logo from '../Logo';
 import { SystemState } from '../../redux/system/types';
+import apiService, { Request } from '../../service/api-service';
 import history from '../History';
-import { Link } from 'react-router-dom';
+import Logo from '../Logo';
 
 interface Props {
   system: SystemState;
   logIn: typeof logIn;
 }
-interface State {}
+
+interface State {
+  email: string;
+  password: string;
+  invalidCredentials: boolean;
+  invalidResponse: boolean;
+  invalidEmail: boolean;
+  missingInputs: boolean;
+}
 
 class LogIn extends React.Component<Props, State> {
+  public state: State = {
+    email: '',
+    password: '',
+    invalidCredentials: false,
+    invalidResponse: false,
+    invalidEmail: false,
+    missingInputs: false
+  };
+
   constructor(props: Props) {
     super(props);
     this.logInNow = this.logInNow.bind(this);
   }
 
-  public logInNow(event: React.SyntheticEvent) {
-    this.props.logIn();
+  public handleChange = (e: React.BaseSyntheticEvent) => {
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
+    // using the "any" type.
+    this.setState<any>({
+      [e.target.id]: e.target.value
+    });
+  };
+
+  public logInNow(event: React.BaseSyntheticEvent) {
     event.preventDefault();
     event.stopPropagation();
-    history.push('/oeci');
+
+    if (this.validateForm()) {
+      const request: Request = {
+        url: '/api/auth_token',
+        data: {
+          email: this.state.email.toLowerCase().trim(),
+          password: this.state.password.trim()
+        },
+        method: 'post'
+      };
+
+      apiService(request)
+        .then(response => {
+          // attach token to auth headers
+          this.props.logIn(response.data.auth_token);
+          history.push('/oeci');
+        })
+        .catch(error => {
+          // ERROR: email and password do not match
+          this.setState({
+            invalidCredentials: true
+          });
+        });
+    } else {
+      // ERRORS: will be set by this.validateForm() and request is not submitted
+    }
   }
 
+  public isEmailValid = (email: string) => {
+    // returns true if the email is correct format: https://www.w3resource.com/javascript/form/email-validation.php
+    return /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email);
+  };
```

Comment:
I didn't see that there was already something in place in ForgotPassword, abstracted email-validation into it's own service and we are now importing that for both components.

It returns false on an empty string so I refactored to have:
`{missingPassword, invalidEmail}`
instead of:
`{missingInputs, invalidEmail}`